apps/chat/agent/tools.py

```diff
@@ -32,44 +37,55 @@ def action(*args, **kwargs):
         raise Exception("Not implemented")
 
 
-class CurrentDatetimeTool(CustomBaseTool):
-    name = "current_datetime"
-    description = "useful for getting the current datetime and timezone in ISO 8601. this should be called before setting any reminders"
+class RecurringReminderTool(CustomBaseTool):
+    name = "recurring-reminder"
+    description = "useful to schedule recurring reminders"
+    requires_session = True
+    args_schema: Type[schemas.RecurringReminderSchema] = schemas.RecurringReminderSchema
 
-    def action(self, *args, **kwargs):
-        timezone = pytz.timezone("Africa/Johannesburg")
-        current_datetime = datetime.now().astimezone(timezone)
-        iso_format = current_datetime.strftime("%Y-%m-%dT%H:%M:%S%z")
-        return str(iso_format)
+    def action(
+        self,
+        datetime_due: datetime,
+        datetime_end: datetime,
+        every: int,
+        period: str,
+        message: str,
+        **kwargs,
+    ):
+        interval_schedule, _created = IntervalSchedule.objects.get_or_create(every=every, period=period)
+        channel_session = ChannelSession.objects.filter(experiment_session=self.experiment_session).first()
+        PeriodicTask.objects.create(
+            name=f"reminder-{self.experiment_session.id}-{uuid.uuid4()}",
+            task=BOT_MESSAGE_FOR_USER_TASK,
+            kwargs=json.dumps({"chat_ids": [channel_session.external_chat_id], "message": message}),
+            start_time=datetime_due,
+            expires=datetime_end,
+            interval=interval_schedule,
+        )
```

Comment:
this could be extracted to a function which would make it easier to read

Reply:
https://github.com/dimagi/open-chat-studio/pull/35/commits/64bf25d03de4c2ca992939257055abd0e457a4e3